src/test/java/com/synacy/lesson02/exercises/EnrollmentServiceTest.java

```diff
@@ -0,0 +1,100 @@
+package com.synacy.lesson02.exercises;
+
+import com.synacy.lesson02.exercises.domain.*;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+/**
+ * Created by banjoe on 5/3/17.
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class EnrollmentServiceTest {
+
+    private EnrollmentService enrollmentService;
+
+    private EnrollmentDto enrollmentDetails;
+    private Student student;
+    private StudentProfile studentProfile;
+    private Set<CourseClass> enrolledClasses;
+    private CourseClass courseClass1;
+    private CourseClass courseClass2;
+
+    @Mock StudentProfileService studentProfileService;
+    @Mock CourseClassService courseClassService;
+    @Mock EnrollmentNotificationService enrollmentNotificationService;
+    @Mock StudyLoadFormatter studyLoadFormatter;
+    @Mock SystemService systemService;
+
+    @Before
+    public void setup() throws Exception {
+        enrollmentService = new EnrollmentService();
+        enrollmentService.setCourseClassService(courseClassService);
+        enrollmentService.setEnrollmentNotificationService(enrollmentNotificationService);
+        enrollmentService.setStudentProfileService(studentProfileService);
+        enrollmentService.setStudyLoadFormatter(studyLoadFormatter);
+        enrollmentService.setSystemService(systemService);
+
+        enrollmentDetails = mock(EnrollmentDto.class);
+        student = mock(Student.class);
+        when(enrollmentDetails.getStudent()).thenReturn(student);
+
+        studentProfile = mock(StudentProfile.class);
+        when(enrollmentDetails.getStudentProfile()).thenReturn(studentProfile);
+
+        courseClass1 = mock(CourseClass.class);
+        courseClass2 = mock(CourseClass.class);
+
+        enrolledClasses = new HashSet<>();
+        enrolledClasses.add(courseClass1);
+        enrolledClasses.add(courseClass2);
+        when(enrollmentDetails.getEnrolledClasses()).thenReturn(enrolledClasses);
```

Comment:
To clear things up, it doesn't mean that just because the test methods are testing multiple concepts within one scenario only that you can put all mocking classes on the setup method. Basically, it is perfectly valid; however, this is how we want to use the setup method. Most of the time, we only want dependency injections hare and each mocking should be done within their respective methods - that is, you have to copy each of these code segments to every test methods here.

In reality, this `EnrollmentService` would certainly have several methods which could spawn several scenarios depending on the various combinations of input. Doing this practice will eventually clutter your test class with several objects which are used in different test methods. This would not be preferable as the test class becomes harder to manage.